File: app/src/main/java/uk/co/sundroid/activity/location/AbstractLocationActivity.java
package uk.co.sundroid.activity.location;

import uk.co.sundroid.AbstractActivity;
import android.content.Intent;
import android.os.Bundle;

public abstract class AbstractLocationActivity extends AbstractActivity {
	
    @Override
    public void onCreate(Bundle savedInstanceState) {
    	super.onCreate(savedInstanceState);
        setContentView(getLayout());
        setActionBarTitle(getViewTitle());
        setDisplayHomeAsUpEnabled();
    }
    
    protected abstract String getViewTitle();
    
    protected abstract int getLayout();

    @Override
    protected void onNavBackSelected() {
        setResult(LocationSelectActivity.RESULT_CANCELLED);
        finish();
    }

	@Override
	protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
		if (resultCode == LocationSelectActivity.RESULT_LOCATION_SELECTED || resultCode == TimeZonePickerActivity.RESULT_TIMEZONE_SELECTED) {
			setResult(LocationSelectActivity.RESULT_LOCATION_SELECTED);
			finish();
		}
	}
    
}

File: app/src/main/kotlin/uk/co/sundroid/activity/location/AbstractLocationActivity.kt
package uk.co.sundroid.activity.location

import uk.co.sundroid.AbstractActivity
import android.content.Intent
import android.os.Bundle

abstract class AbstractLocationActivity : AbstractActivity() {

    protected abstract val viewTitle: String

    protected abstract val layout: Int

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layout)
        setActionBarTitle(viewTitle)
        setDisplayHomeAsUpEnabled()
    }

}
